src/Microsoft.IdentityModel.Protocols/Configuration/HttpDocumentRetriever.cs

```diff
@@ -76,7 +70,14 @@ public bool RequireHttps
             try
             {
                 IdentityModelEventSource.Logger.WriteVerbose(LogMessages.IDX10805, address);
-                HttpResponseMessage response = await _httpClient.GetAsync(address, cancel).ConfigureAwait(false);
+                HttpResponseMessage response;
+
+                if (_httpClient == null)
+                    using (HttpClient _httpClient = new HttpClient())
```

Comment:
FWIW, this pattern looks weird and is rather inefficient, as `HttpClient` has been specifically designed to be cached and reused concurrently.

Instead, why not adding a static readonly `DefaultHttpClient` field and use it if no custom client has been provided? This way, you wouldn't have to dispose anything at all.